noty-api/data/src/dev/shreyaspatil/noty/data/database/NotyDatabase.kt

```diff
@@ -25,17 +27,24 @@ import org.jetbrains.exposed.sql.transactions.transaction
 /**
  * Initializes dev.shreyaspatil.noty.data.database connection with application
  */
-fun initDatabase(databaseConfig: DatabaseConfig) {
+fun initDatabase(hikariConfig: HikariConfig) {
     val tables = arrayOf(Users, Notes)
 
-    Database.connect(
-        url = with(databaseConfig) { "jdbc:postgresql://$host:$port/$name" },
-        driver = "org.postgresql.Driver",
-        user = databaseConfig.user,
-        password = databaseConfig.password
-    )
+    Database.connect(hikari(hikariConfig))
 
     transaction {
         SchemaUtils.createMissingTablesAndColumns(*tables)
     }
+}
+
+private fun hikari(hikariConfig: HikariConfig): HikariDataSource {
```

Comment:
1. Rename this function to `createDataSource`.
2. Change the return type to `DataSource` which is the super interface of all common Data Sources.

```suggestion
private fun createDataSource(databaseConfig: DatabaseConfig): DataSource {
```

Reply:
Renamed it as per suggestion.